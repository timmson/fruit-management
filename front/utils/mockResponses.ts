export function okAndJson(json: any): Promise<Response> {
    return Promise.resolve({
        arrayBuffer(): Promise<ArrayBuffer> {
            return Promise.resolve(undefined);
        },
        blob(): Promise<Blob> {
            return Promise.resolve(undefined);
        },
        body: undefined,
        bodyUsed: false,
        clone(): Response {
            return undefined;
        },
        formData(): Promise<FormData> {
            return Promise.resolve(undefined);
        },
        headers: undefined,
        json(): Promise<any> {
            return Promise.resolve(json);
        },
        ok: true,
        redirected: false,
        status: 200,
        statusText: "OK",
        text(): Promise<string> {
            return Promise.resolve("");
        },
        type: undefined,
        url: ""
    })
}