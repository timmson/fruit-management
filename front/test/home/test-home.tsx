import React from "react"
import renderer from "react-test-renderer"
import Context from "../../src/context"
import {okAndJson} from "../../utils/mockResponses"
import Home from "../../src/home/home"

global.fetch = jest.fn(() => okAndJson({}))

describe("Plan should", () => {

	test("return plan form", () => {
		const context = []

		const component = renderer.create(
			<Context.Provider value={context}>
				<Home/>
			</Context.Provider>
		)

		expect(component.toJSON()).toMatchSnapshot()

		component.unmount()
	})

})