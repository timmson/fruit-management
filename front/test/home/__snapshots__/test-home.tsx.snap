// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan should return plan form 1`] = `
[
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h3>
        Моя активность
      </h3>
      <table
        className="basic w-100"
      >
        <thead>
          <tr
            className="fw-bold"
          >
            <th>
              Задача
            </th>
            <th>
              Статус
            </th>
            <th>
              ПН
            </th>
            <th>
              ВТ
            </th>
            <th>
              СР
            </th>
            <th>
              ЧТ
            </th>
            <th>
              ПТ
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="fw-bold"
          >
            <td
              className="text-end"
              colSpan="2"
            >
              Итого
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
            <td
              className="text-center"
            >
              0
            </td>
          </tr>
        </tbody>
      </table>
      <br />
      <h3>
        Мои подписки
      </h3>
      <table
        className="basic w-100"
      >
        <thead>
          <tr
            className="fw-bold"
          >
            <td>
              Задача
            </td>
            <td>
              Пользователь
            </td>
            <td>
              %
            </td>
            <td>
              Статус
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      className="col-6"
    >
      <h3>
        Мои задачи
      </h3>
      <table
        className="basic w-100"
      >
        <thead>
          <tr
            className="fw-bold"
          >
            <td>
              Задача
            </td>
            <td>
              Состояние
            </td>
            <td>
              %
            </td>
            <td>
              Приоритет
            </td>
            <td>
              Статус
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <h3>
        Календарь
      </h3>
      <table
        className="basic w-100"
      >
        <thead>
          <tr>
            <th
              rowSpan="2"
            >
              Задача
            </th>
            <th
              rowSpan="2"
            >
              Длина
            </th>
            <th
              colSpan={0}
            >
              7
            </th>
          </tr>
          <tr />
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <h3>
        Активность команды
      </h3>
      <table
        className="basic"
      >
        <tbody>
          <tr>
            <td
              className="text-start"
            >
              Пока здесь пусто;(
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
